tests/testthat/test-driver-oracle.R

```diff
@@ -18,3 +19,19 @@ test_that("can detect existence of table", {
   tbl2 <- local_table(con, "mtcars_test", mtcars)
   expect_true(dbExistsTable(con, tbl2))
 })
+
+test_that("Writing date/datetime with batch size > 1", {
+  # See #349, #350, #391
+  con <- test_con("ORACLE")
+
+  ir <- datasets::iris
+  values <- data.frame(
+    datetime = as.POSIXct(as.numeric(ir$Petal.Length * 10), origin = "2024-01-01", tz = "UTC"),
+    date    = as.Date(as.numeric(ir$Petal.Length * 10), origin = "2024-01-01", tz = "UTC"),
+    integer = as.integer(ir$Petal.Width * 100),
+    double = ir$Sepal.Length,
+    varchar = ir$Species)
```

Comment:
```suggestion
  values <- data.frame(
    datetime = as.POSIXct(as.numeric(iris$Petal.Length * 10), origin = "2024-01-01", tz = "UTC"),
    date    = as.Date(as.numeric(iris$Petal.Length * 10), origin = "2024-01-01", tz = "UTC"),
    integer = as.integer(iris$Petal.Width * 100),
    double = iris$Sepal.Length,
    varchar = iris$Species)
```

Objects from datasets are available without namespacing!